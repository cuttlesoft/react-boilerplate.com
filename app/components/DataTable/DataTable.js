/* eslint-disable react/prop-types */
import React from 'react'
import { DataTable } from 'grommet'

import { doc } from './DataTable.doc'
import helpers from '../../utils/helpers'

/**
 *
 * DataTable
 *
 */
function createDataTable({
  a11yTitle,
  alignSelf,
  columns,
  data,
  groupBy,
  margin,
  onSearch,
  resizeable,
  size,
  sortable,
  step,
  ...rest
}) {
  return (
    <DataTable
      a11yTitle={a11yTitle}
      alignSelf={alignSelf}
      columns={columns}
      data={data}
      groupBy={groupBy}
      margin={margin}
      onSearch={onSearch}
      resizeable={resizeable}
      size={size}
      sortable={sortable}
      step={step}
      {...rest}
    />
  )
}

export default helpers.createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: createDataTable,
})
