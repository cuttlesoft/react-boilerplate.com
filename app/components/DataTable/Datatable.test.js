import React from 'react'
import { render } from '@testing-library/react'

import DataTable from './DataTable'

const renderComponent = (props = {}) => render(<DataTable {...props} />)

/**
 *
 * Tests for DataTable
 *
 */
describe('DataTable', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })
})
