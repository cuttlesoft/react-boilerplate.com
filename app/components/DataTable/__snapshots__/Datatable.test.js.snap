// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable renders and matches snapshot 1`] = `
<div>
  <table
    class="StyledTable-sc-1m3u5g-6 bJymoW StyledDataTable-xrlyjm-0 hJcOxq"
  >
    <thead
      class="StyledTable__StyledTableHeader-sc-1m3u5g-4 kdPLsZ StyledDataTable__StyledDataTableHeader-xrlyjm-3 hOGMbn"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2 dSAyNg StyledDataTable__StyledDataTableRow-xrlyjm-1 jKzMJr"
      />
    </thead>
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3 bMdIOJ StyledDataTable__StyledDataTableBody-xrlyjm-2 kERYjL"
    />
  </table>
</div>
`;
